### PR TITLE
sci-physics/thepeg: add 2.3.0

### PR DESCRIPTION
**Pull request by APN-Pucky**
PR created by [pkgpr](https://github.com/APN-Pucky/pkgpr).

### CODE REVIEW

File: sci-physics/thepeg/metadata.xml [hunk 1]
```diff
@@ -14,8 +18,9 @@
 	</longdescription>
 	<use>
 		<flag name="fastjet">Adds support for <pkg>sci-physics/fastjet</pkg></flag>
-		<flag name="hepmc">Adds support for High Energy Physics Monte Carlo
-		Generators <pkg>sci-physics/hepmc</pkg></flag>
+		<flag name="hepmc">Adds support for High Energy Physics Monte Carlo Generators<pkg>sci-physics/hepmc</pkg></flag>
+		<flag name="hepmc3">Adds support for High Energy Physics Monte Carlo Generators 3<pkg>sci-physics/hepmc</pkg></flag>
 		<flag name="lhapdf">Adds support for <pkg>sci-physics/lhapdf</pkg></flag>
+		<flag name="rivet">Adds support for <pkg>sci-physics/rivet</pkg></flag>
 	</use>
```
Comment (Jamim) [inline sci-physics/thepeg/metadata.xml:25]:
```suggestion
	</use>
	<upstream>
		<doc>https://thepeg.hepforge.org/doxygen/</doc>
	</upstream>
```

File: sci-physics/thepeg/thepeg-2.3.0.ebuild [hunk 2]
```diff
@@ -0,0 +1,111 @@
...
+# Distributed under the terms of the GNU General Public License v2
+
+EAPI=8
+
+inherit autotools elisp-common java-pkg-opt-2
+
+MY_P=ThePEG-${PV}
+
+DESCRIPTION="Toolkit for High Energy Physics Event Generation"
+HOMEPAGE="http://home.thep.lu.se/ThePEG/"
```
Comment (Jamim) [inline sci-physics/thepeg/thepeg-2.3.0.ebuild:11]:
```suggestion
HOMEPAGE="http://home.thep.lu.se/ThePEG/ https://thepeg.hepforge.org/"
```

**Review by Jamim — CHANGES_REQUESTED** [review]

File: sci-physics/thepeg/thepeg-2.3.0.ebuild (hunk 2 above)
Comment (APN-Pucky) [reply sci-physics/thepeg/thepeg-2.3.0.ebuild]: Thanks, I squashed it in the first commit and also changed the first HOMEPAGE to https.

File: sci-physics/thepeg/thepeg-2.3.0.ebuild [hunk 3]
```diff
@@ -0,0 +1,114 @@
...
+SRC_URI="https://www.hepforge.org/archive/thepeg/${MY_P}.tar.bz2
+	test? (
+		hepmc3? (
+			${TEST_URI}/cteq6ll.LHpdf
+			${TEST_URI}/cteq5l.LHgrid
+			${TEST_URI}/GRV98nlo.LHgrid
+			${TEST_URI}/MRST2001nlo.LHgrid )
+	)"
+
+LICENSE="GPL-2"
```
Comment (Jamim) [inline sci-physics/thepeg/thepeg-2.3.0.ebuild:26]: It's `GPL-3` now. Please see the [COPYING](https://phab.hepforge.org/source/thepeghg/browse/release-2-3/COPYING) and [README](https://phab.hepforge.org/source/thepeghg/browse/release-2-3/README).
```suggestion
LICENSE="GPL-3"
```

**Review by Jamim — CHANGES_REQUESTED** [review]

File: sci-physics/thepeg/thepeg-2.3.0.ebuild (hunk 3 above)
Comment (APN-Pucky) [reply sci-physics/thepeg/thepeg-2.3.0.ebuild]: Previous versions 2.2.2 and 2.2.3 are also GPL-3 -> updated.

File: sci-physics/thepeg/thepeg-2.3.0.ebuild [hunk 4]
```diff
@@ -0,0 +1,114 @@
...
+	java? ( virtual/jdk:1.8 )
+	test? (
+		sys-process/time
+		dev-libs/boost
+	)"
+RDEPEND="${CDEPEND}
+	java? ( virtual/jre:1.8 )
+"
+
+S="${WORKDIR}/${MY_P}"
```
Comment (Jamim) [inline sci-physics/thepeg/thepeg-2.3.0.ebuild:51]:
```shell
# pkgcheck scan thepeg-2.3.0.ebuild 
sci-physics/thepeg
  VariableOrderWrong: version 2.3.0: variable S should occur before RESTRICT
```
[pkgcheck](https://github.com/pkgcore/pkgcheck) will keep complaining until you move `S` before `LICENSE`.

```suggestion
S="${WORKDIR}/${MY_P}"

LICENSE="GPL-3"
SLOT="0/30"
KEYWORDS="~amd64"
IUSE="emacs fastjet +hepmc3 lhapdf static-libs test zlib rivet"
RESTRICT="!test? ( test )"

CDEPEND="
	sci-libs/gsl:0=
	emacs? ( >=app-editors/emacs-23.1:* )
	fastjet? ( sci-physics/fastjet:0= )
	rivet? ( sci-physics/rivet:3=[hepmc3] )
	hepmc3? ( sci-physics/hepmc:3= )
	lhapdf? ( >=sci-physics/lhapdf-6.0:0= )
	zlib? ( sys-libs/zlib:0= )"
DEPEND="${CDEPEND}
	sci-libs/gsl:=
	java? ( virtual/jdk:1.8 )
	test? (
		sys-process/time
		dev-libs/boost
	)"
RDEPEND="${CDEPEND}
	java? ( virtual/jre:1.8 )
"
```

**Review by Jamim — CHANGES_REQUESTED** [review]

File: sci-physics/thepeg/thepeg-2.3.0.ebuild (hunk 4 above)
Comment (APN-Pucky) [reply sci-physics/thepeg/thepeg-2.3.0.ebuild:51]: I was still on pkgcheck-0.10.27 which did not complain.

**Review by Jamim — APPROVED** [review]
Thank you for your contribution, @APN-Pucky!

File: sci-physics/thepeg/thepeg-2.3.0.ebuild (hunk 2 above)
Comment (Jamim) [reply sci-physics/thepeg/thepeg-2.3.0.ebuild]: Unfortunately, changing the first link to `https` prevents access due to [a certificate mismatch issue](https://www.ssllabs.com/ssltest/analyze.html?d=home.thep.lu.se).

Comment (APN-Pucky) [reply sci-physics/thepeg/thepeg-2.3.0.ebuild]: Good that it works now at least, I dropped the `https` for now.

**Review by AndrewAmmerlaan — COMMENTED** [review]
Version 2.2.2 uses hepmc:2, but 2.2.3 uses hepmc:3 (both have the same `hepmc` flag).

I think we should either keep the name of the flag the same in 2.3.0. Or adjust the name of the flag in 2.2.3 for consistency.

Otherwise LGTM

**Comment by APN-Pucky** [comment]
`hepmc3` is also used by pythia and rivet. So I decided to go with that.

File: sci-physics/thepeg/thepeg-2.2.3-r1.ebuild [hunk 5]
```diff
@@ -22,14 +22,14 @@ S="${WORKDIR}/${MY_P}"
 LICENSE="GPL-3"
 SLOT="0/30"
 KEYWORDS="~amd64 ~x86"
-IUSE="emacs fastjet hepmc java lhapdf static-libs test zlib"
+IUSE="emacs fastjet hepmc3 java lhapdf static-libs test zlib"
```
Comment (AndrewAmmerlaan) [inline sci-physics/thepeg/thepeg-2.2.3-r1.ebuild:25]: In 2.3.0 this flag is enabled by default, lets do that here as well to also avoid surprises due to the renaming here.

**Comment by AndrewAmmerlaan** [comment]
Thanks :+1: 

```
IncorrectCopyright: version 2.2.2-r1: incorrect copyright year 2022: '# Copyright 1999-2022 Gentoo Authors'
IncorrectCopyright: version 2.2.3-r1: incorrect copyright year 2022: '# Copyright 1999-2022 Gentoo Authors'
```

Since we are touching the older ebuilds as well, we have to update the copyright header. I fixed this in a follow-up commit.

**Pull request closed without merging**